core/trino-main/src/main/java/io/trino/metadata/LanguageFunctionManager.java

```diff
@@ -428,6 +440,24 @@ private synchronized void analyzeAndPlan(AccessControl accessControl)
                 analyzing = false;
             }
 
+            private SqlRoutineAnalysis analyze(FunctionContext context)
+            {
+                try {
+                    return analyzer.analyze(context.session(), context.accessControl(), functionSpecification);
+                }
+                catch (TrinoException e) {
+                    if (identityLoader.isEmpty() || (e instanceof LanguageFunctionAnalysisException)) {
+                        throw e;
```

Comment:
What is the `identityLoader.isEmpty()` condition for?
Unless this is obvious to the reader, add a code comment

Reply:
I extracted this to a method with a comment.